lib/reverse_sentence.rb

```diff
@@ -1,6 +1,34 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n), where n is the length of a sentence.
+# Space complexity: O(n)
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  return false if my_sentence == nil || my_sentence.length == 0
+  word_start = 0
+  word_end = 0
+  i = 0
+  new_sentence = ""
+ 
+  while i < my_sentence.length
+    while i < my_sentence.length && my_sentence[i] == " "
+      new_sentence = my_sentence[i] + new_sentence
+      i += 1
+    end
+    word_start = i
+
+    while i < my_sentence.length && my_sentence[i] != " "
+      i += 1
+    end
+    word_end = i - 1
+
+    while word_start <= word_end
```

Comment:
Style nit: this looks like it's reversing a string. Moving this into a helper method might make things easier to read. 🙂